src/protocol/receiver/time.rs

```diff
@@ -0,0 +1,111 @@
+use std::time::{Duration, Instant};
+
+use stats::OnlineStats;
+
+use crate::protocol::{TimeBase, TimeSpan, TimeStamp};
+
+pub(crate) struct SynchronizedRemoteClock {
+    tolerance: Duration,
+    time_base: TimeBase,
+    stats: Option<OnlineStats>,
+}
+
+impl SynchronizedRemoteClock {
+    const MAX_SAMPLES: usize = 1_000;
+    const DRIFT_TOLERANCE: Duration = Duration::from_millis(5);
+
+    pub fn new(now: Instant) -> Self {
+        Self {
+            // TODO: Drift deviation tolerance should be parameterized.
+            //       It wasn't in the reference implementation, but I added it because the reference
+            //       implementation is susceptible to invalid clock adjustments during periods of
+            //       acute network latency
```

Comment:
What is the tolerance in the reference impl?

Reply:
It has no tolerance, it adjusts the time without regard to drift deviation.